### ISSUE TITLE
can i change the __C.TRAIN.RPN_BATCHSIZE = 256 ?

### ISSUE THREAD

**Issue by devendraswamy** [jwyang/faster-rcnn.pytorch#761]

**Comment by loolzaaa** [comment]
I think yes.
This number = maximum proposals (foreground + background) in anchor target layer, but change this parameter didn't get significant improvement of net metrics for me.

Also, look at the __C.TRAIN.RPN_FG_FRACTION parameter (default = 0.5).

**Comment by devendraswamy** [comment]
Thank you for the valuable information , one more doubt 
can i change this parameter in config.py ?


# Minibatch size (number of regions of interest [ROIs])
--
 __C.TRAIN.BATCH_SIZE = 128

**Comment by devendraswamy** [comment]
Actually my problem is , when i given the large dataset to model then my system was restarted , so i want to reduce the data-loading  ?

**Comment by loolzaaa** [comment]
> an i change this parameter in config.py ?
> __C.TRAIN.BATCH_SIZE = 128

This parameter = maximum proposals (foreground + background) in **proposal** target layer.
Also, there is fraction for it.

> Actually my problem is , when i given the large dataset to model then my system was restarted , so i want to reduce the data-loading ?

**What you mean, when say "large dataset"?**

Parameters above are not responsible for dataset or batch size for model. Actually:
* Batch size for model = number of images processed through model in one forward pass.
* Batch size in anchor/proposal target layer = number of proposals in one image. This parameters doesn't need more memory.

**Comment by devendraswamy** [comment]
Thank you but could you please tell me the reason for system restarting ?

**Comment by loolzaaa** [comment]
Is there some trigger for restarting? High temperature of CPU of GPU? If you see BSOD before reboot, this can be RAM problem.

Need more info about your situation.

In which cases you system is going to reboot, in which not...

**Comment by devendraswamy** [comment]
when I am given large data to model then that time that will restarted and model working fine for less data. I am successfully train the model with pascal and coco datsets and custom data also.

**Comment by loolzaaa** [comment]
> I am **successfully** train the model with pascal and coco datsets and custom data also.

COCO dataset successfully trained? Then what do you mean when say "large data"? Many pictures? Big resolution of pictures... etc...?

**Comment by devendraswamy** [comment]
I mean that will restart when I am giving my own dataset ( more images as
compared to coco)

**Comment by loolzaaa** [comment]
You need to check GPU/RAM memory while training (before reboot).

Try to increase RAM swap file size.

**Comment by devendraswamy** [comment]
Thank you for your suggestion , I will check it and let you know.

**Comment by devendraswamy** [comment]
Hi..
one more help ,I want to decrease the loss of the model , so which parameters i have to change ?

**Comment by loolzaaa** [comment]
loss of the model is the main indicator, almost all parameters can increase or decrease loss.
If you talk about train this network on default datasets like voc or coco then cfg from lib is the best choice, no need to change something.
If you talk about train this network on custom dataset then look into optimizer parameters (lr, weight decay). Also, look into learning rate annealing (decrease LR every X epochs - `adjust_learning_rate` method). NMS parameters also may affect on loss. Try to change RPN anchors and RATIOS to another value.

**Anyway, you need some metrics to rate success of your model, not only loss.**

**Comment by devendraswamy** [comment]
could you provide good documentation for FRCNN model ?

**Comment by loolzaaa** [comment]
> Why coco using four anchor boxes[4,8,16,32]

More image size need more different anchors

> could you provide good documentation for FRCNN model ?

[Blog](http://www.telesens.co/2018/03/11/object-detection-and-classification-using-r-cnns/) from [ankur6ue](https://github.com/ankur6ue)

**Comment by devendraswamy** [comment]
Thank  you !

**Comment by devendraswamy** [comment]
Hi..

In this project , max gt boxes = 20 for pascal_voc and max gt boxes = 50 for coco. can i change that one because in my data max bounding boxes for one image is 270(approximately equal to 270).

please help , Thank you in advance.

**Comment by loolzaaa** [comment]
Yes, you can change it. Also, if you know maximum size of bounding box, may be you want to change RATIOS and SCALES for anchor generation layer.

**Comment by devendraswamy** [comment]
How can I calculated that one ?

**Comment by devendraswamy** [comment]
I thought my data will had max gt bounding boxes size is 270.

**Comment by loolzaaa** [comment]
Hmm... some confusion...
`max gt boxes` parameter = maximum **number** of ground-truth boxes per image, NOT size of boxes.

**Comment by devendraswamy** [comment]
Sry .... Yes my images had max no of gt boxes are 270.

**Comment by devendraswamy** [comment]
any calculations for changing anchor boxes and anchor ratios ? how can i define those parameters based upon data ?

**Comment by loolzaaa** [comment]
You need to read manual carefully, especially "anchor generation layer" part.

**Comment by devendraswamy** [comment]
Thank you so much
